test/extended/router/gatewayapicontroller.go

```diff
@@ -539,3 +591,74 @@ func assertHttpRouteSuccessful(oc *exutil.CLI, gwName, name string) (*gatewayapi
 	e2e.Logf("httpRoute %s/%s successful", namespace, name)
 	return checkHttpRoute, nil
 }
+
+// assertHttpRouteConnection checks if the http route of the given name replies successfully,
+// and returns an error if not
+func assertHttpRouteConnection(oc *exutil.CLI, hostname, gwName string) error {
+	gwapiClient := gatewayapiclientset.NewForConfigOrDie(oc.AdminConfig())
+	ingressNameSpace := "openshift-ingress"
+	gateway, errGwStatus := gwapiClient.GatewayV1().Gateways(ingressNameSpace).Get(context.TODO(), gwName, metav1.GetOptions{})
+	if errGwStatus != nil || gateway == nil {
+		e2e.Failf("Unable to create httpRoute, no gateway available during route assertion %v", errGwStatus)
+	}
+
+	// Create the http client to check the header.
+	client := &http.Client{
+		Timeout: 10 * time.Second,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+
+	// Make sure the DNSRecord is ready to use.
+	assertDNSRecordStatus(oc, gwName, "", "")
+
+	// Wait and check that the dns name resolves first. Takes a long time, so
+	// if the hostname is actually an IP address, skip this.
+	dnsResolutionTimeout := 10 * time.Minute
+	if net.ParseIP(hostname) == nil {
```

Comment:
We cannot have an IP here as we pass a hostname of the route, not service's LB target which may be an IP for some cloud providers. So, let's just remove this check.

Reply:
done